hatchways/team-bond
# In/esteban/profile listings page #48

### What this PR does (required):
- Front end changes
  - refactored some components to child components
  - extract profile list items into their own components calling a parent function only when clicked
  - pass data to profileSearch component and have it call the parent only when search button is clicked.
  - use 2 datetime input components in the profileSearch component instead of city/date, per specs
  - integrate the api call to the backend - makes use of encoded query params
 - Backend
   - add some seed data for a demo sitter with an availability
   - repurposed the user search controller to get available profiles
   - added new blurb property in the sitter schema
   - changed some fields to not required

### Screenshots / Videos (front-end only):
set filters
<img width="956" alt="image" src="https://user-images.githubusercontent.com/26440947/158974658-4615fa1b-2840-43ac-b795-82016d79ca1b.png">

get data from nack-end
<img width="820" alt="image" src="https://user-images.githubusercontent.com/26440947/158982290-d713dedc-4ad6-437d-b545-a672fb069373.png">


### Any information needed to test this feature (required):
- log in with the demo user
- go to `http://localhost:3000/listings` and select date-times between 9am and 5pm for any date.
- expect to see the demo sitter profile in the screenshot above

### Any issues with the current functionality (optional):
- Only searching by the day of the week extracted from the selected date
- only works if both from/to date-times are in the same day.
- City is not part of the schema. Currently it's extracted from the address line
